lightning/src/ln/channelmanager.rs

```diff
@@ -2292,48 +2305,57 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 			if let &PendingHTLCRouting::Forward { ref short_channel_id, .. } = routing {
 				let id_option = channel_state.as_ref().unwrap().short_to_id.get(&short_channel_id).cloned();
 				if let Some((err, code, chan_update)) = loop {
-					let forwarding_id = match id_option {
+					let forwarding_id_opt = match id_option {
 						None => { // unknown_next_peer
-							break Some(("Don't have available channel for forwarding as requested.", 0x4000 | 10, None));
+							if fake_scid::is_valid_phantom(self.fake_scid_offset, *short_channel_id) {
```

Comment:
If you take my suggestion of making the phantom scids have a random offset, can you add a comment here noting this is probably a timing oracle as to whether an scid is a phantom?